src/fty_srr.cc

```diff
@@ -35,12 +36,15 @@
 //functions
 
 void usage();
-volatile bool g_compute = true;
+volatile bool g_exit = false;
+std::condition_variable g_cv;
+std::mutex g_cvMutex;
 
-void sigHandler(int s)
+void sigHandler(int )
 {
-    log_debug("Caught signal %d\n", s);
-    g_compute = false;
+    log_debug("Interrupt signal");
```

Comment:
Refrain from doing any work in the signal handler that absolutely **cannot** be deferred. The process is not guaranteed to be in a quiescent state.